packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -77,25 +104,175 @@ function WorklistDataGrid(props) {
     rows,
     columnHeaderHeight,
     onColumnSelect,
+    onCellSelect,
+    onRowSelect,
+    onRowSelectAll,
+    onClearSelectedRows,
+    onDisableSelectableRows,
+    hasSelectableRows,
     intl,
+    rowHeaderIndex,
   } = props;
 
+  const grid = useRef();
   const focusedRow = useRef(0);
   const focusedCol = useRef(0);
-  const grid = useRef();
+  const ariaLiveMsg = useRef();
+
+  const [currentSelectedCell, setCurrentSelectedCell] = useState(null);
+  const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(columns);
+
+  const theme = useContext(ThemeContext);
+
+  const isRowSelectionCell = (columnIndex) => (
+    hasSelectableRows && columnIndex < displayedColumns.length && displayedColumns[columnIndex].id === WorklistDataGridUtils.ROW_SELECTION_COLUMN.id
+  );
 
-  const gridRef = useCallback((node) => {
-    grid.current = node;
-    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
+  const isCellSelected = (rowId, columnId) => (currentSelectedCell && currentSelectedCell.rowId === rowId && currentSelectedCell.columnId === columnId);
+
+  const setFocus = (rowIndex, colIndex, makeActiveElement) => {
+    let focusedCell = grid.current.rows[rowIndex].cells[colIndex];
+    if (isRowSelectionCell(colIndex) && focusedCell.firstChild) {
+      focusedCell = focusedCell.firstChild;
+    }
     focusedCell.tabIndex = 0;
-  }, []);
+    if (makeActiveElement && focusedCell.focus) {
+      focusedCell.focus();
+    }
+  };
 
-  const handleKeyDown = (event) => {
-    const currentFocusedRow = event.target.parentElement.rowIndex;
-    const currentFocusedCol = event.target.cellIndex;
+  const removeTabStop = (rowIndex, colIndex) => {
+    const cell = grid.current.rows[rowIndex].cells[colIndex];
+    // Remove Tab stop from previous cell in table that has focus and set it to the
+    // cell that was clicked.
+    cell.tabIndex = -1;
+    if (isRowSelectionCell(colIndex) && cell.firstChild) {
+      // Also remove tabstop from the checkbox in the case of a row selection cell.
+      cell.firstChild.tabIndex = -1;
+    }
+  };
+
+  const setFocusedRowCol = (newRowIndex, newColIndex, makeActiveElement) => {
+    removeTabStop(focusedRow.current, focusedCol.current);
+    focusedRow.current = newRowIndex;
+    focusedCol.current = newColIndex;
+    setFocus(newRowIndex, newColIndex, makeActiveElement);
```

Comment:
Question.  Should the Cell components be responsible for setting their own tabIndex and focus?  Would that encapsulate away that functionality and simplify the WorklistDataGrid logic?

Reply:
I do not think so. When navigating, there are two cells involved and since there is no re-rendering and we must maintain only one tab stop, the 'swapping' of the tabstop and focus from one cell to another should be done outside the cell.